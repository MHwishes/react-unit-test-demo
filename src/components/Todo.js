import React from 'react'
import PropTypes from 'prop-types'

const Todo = ({onClick, completed, text}) => {
    return (
        <li
            className={completed ? 'completed' : ''}
        >
            <div className="view">
                <input className="toggle" type="checkbox" onClick={onClick}/>
                <label>{text}</label>
            </div>
        </li>
    )
};

Todo.propTypes = {
    onClick: PropTypes.func.isRequired,
    completed: PropTypes.bool.isRequired,
    text: PropTypes.string.isRequired
};

export default Todo
