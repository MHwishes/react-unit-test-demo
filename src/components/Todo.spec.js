import {shallow} from "enzyme";
import Todo from "./Todo";
import toJson from "enzyme-to-json";
import React from "react";
import configMockStore from 'redux-mock-store';

const mockStore = configMockStore();

describe('shallow render Todo component', () => {
    it('render Todo', () => {
        const initialState = {};
        const store = mockStore(initialState);
        const props = {id: 1, text: "test", completed: false};
        const wrapper = shallow(<Todo store={store} {...props}/>);
        expect(toJson(wrapper.dive())).toMatchSnapshot();
    })
});
