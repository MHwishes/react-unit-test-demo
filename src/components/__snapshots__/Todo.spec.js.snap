// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shallow render Todo component render Todo 1`] = `
<li
  className=""
>
  <div
    className="view"
  >
    <input
      className="toggle"
      onClick={[Function]}
      type="checkbox"
    />
    <label>
      test
    </label>
  </div>
</li>
`;
